frontend/apps/filemanager/filemanagermenu.lua

```diff
@@ -292,16 +351,21 @@ A subfolder will be shortened from:
 `/mnt/onboard/.books/Manga/Cells at Work`
 To:
 `Manga/Cells at Work`.]]),
-            },
-            {
-                text = _("Show filename in Open last/previous menu items"),
-                checked_func = function() return G_reader_settings:isTrue("open_last_menu_show_filename") end,
-                callback = function() G_reader_settings:flipNilOrFalse("open_last_menu_show_filename") end,
-            },
-            {
-                text = _("Auto-remove deleted or purged items from history"),
-                checked_func = function() return G_reader_settings:isTrue("autoremove_deleted_items_from_history") end,
-                callback = function() G_reader_settings:flipNilOrFalse("autoremove_deleted_items_from_history") end,
+                    },
+                    {
+                        text = _("Lock home folder"),
+                        enabled_func = function()
+                            return G_reader_settings:has("home_dir")
+                        end,
+                        checked_func = function()
+                            return G_reader_settings:isTrue("lock_home_folder")
+                        end,
+                        callback = function()
+                            G_reader_settings:flipNilOrFalse("lock_home_folder")
+                            UIManager:broadcastEvent(Event:new("Refresh"))
```

Comment:
There's only one widget that has a onRefresh handler, and it's FileManager - so you could just use like it's done in other actions in this file: `self.ui:onRefresh()`